backend/ASTU-backend-group-1/delivery/controllers/user_controllers.go

```diff
@@ -0,0 +1,155 @@
+package controllers
+
+import (
+	"astu-backend-g1/domain"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+type UserController struct {
+	userUsecase domain.UserUsecase
+}
+
+func NewUserController(userUsecase domain.UserUsecase) *UserController {
+	return &UserController{userUsecase: userUsecase}
+}
+
+func (c *UserController) Register(ctx *gin.Context) {
```

Comment:
Here we need to hash the user password for handling the hashing be sure to do that

Reply:
I already did it in user usecase